nautobot_golden_config/models.py

```diff
@@ -486,21 +495,33 @@ def get_absolute_url(self):  # pylint: disable=no-self-use
 
     def __str__(self):
         """Return a simple string if model is called."""
-        return "Golden Config Settings"
+        return "Configuration Object"
 
     def delete(self, *args, **kwargs):
         """Enforce the singleton pattern, there is no way to delete the configurations."""
 
-    @classmethod
-    def load(cls):
-        """Enforce the singleton pattern, fail it somehow more than one instance."""
-        if len(cls.objects.all()) != 1:
-            raise ValidationError("There was an error where more than one instance existed for a setting.")
-        return cls.objects.first()
+    class Meta:
+        """Set unique fields for model."""
+
+        verbose_name = "Golden Config Setting"
+
+    def save(self, *args, **kwargs):
+        """Overload save and re-assign the first object.pk and enforce creation of 1 object only.
```

Comment:
Why re-assign this? It's not clear to me why this is advantageous over the previous implementation.

Reply:
Previously it did not actually enforce only one object, we were able create N number of instances. And due to delete being overloaded to there was no way to remove. This was discovered when using the ORM and this change enforces 1 object if you touch the ORM.

Reply:
There is no guarantee that when someone touches the orm that they do not provide valid instances, see https://github.com/nautobot/nautobot/issues/1034 as an example. I am all for creating a more robust solution, however, I do not understand why one would need to re-assign an object to do so.

Reply:
This basically forces the integrity error to be raised. After many failed attempts to make the previous solution accomplish the same goal, this seemed to work good. Maybe the wording is the issue here? Nothing actually gets re-assigned, I guess. It's just an attempt, as that primary-key should be the only primary-key, thus raising the error.

Reply:
It may just be a wording thing, I certainly do not understand what it is doing. I've asked the great @lampwins to weigh in here, as he helped me to begin with and has some ideas.

Reply:
I read through a second time and I get Jeremy's point now. We never actually implemented a creation constraint through `save()` so he is right that there is nothing to stop you from creating more than one. There was however a provision that you will always get first one.

That being said, I provide here a (hopefully) more complete singleton example we are playing with for a use case in core. It enforces one instance, but still offers an interface to always get that one instance, or create it if it does not exist:

```python
class SchemaQuerySingleton(BaseModel):
    """
    Singleton model which stores the GraphQL schema query so it can easily be shared between
    running instances/workers as the schema changes.
    """
    pickled_data = DillPickledObjectField()

    def save(self, *args, **kwargs):
        if not self.present_in_database and self.__class__.objects.exists():
            raise SchemaQuerySingletonMultipleObjectsException("SchemaQuerySingleton already exists!")
        super().save(*args, **kwargs)

    @classmethod
    def load(cls, initializer=None):
        """
        Return the singleton instance, or create it if it does not exist by passing a callable
        to populate the schema query data for the first time.

        Args:
            initializer: callable used to populate `pickled_data` for the first time

        Returns: SchemaQuerySingleton instance or None
        """
        obj = cls.objects.first()
        if obj is None and initializer is not None:
            obj = cls.objects.create(pickled_data=initializer())

        return obj
```

You can of course ignore the `initializer` bit as that is unrelated to the actual pattern.

Reply:
Ken is also right about [nautobot/nautobot#1034](https://github.com/nautobot/nautobot/issues/1034), but in this specific case we can take reasonable precautions via `save()`.

Reply:
Yes, as indicated, I am more than happy to provide a more robust solution. The terminology used and my lack of understanding of what was happening was confusing to understand the advantages. 

What say you @whitej6 @h4ndzdatm0ld @mitchell-foxworth, does the pattern @lampwins provide make more or less sense?

Reply:
Without going too far out of scope of what the work of this PR is accomplishing, I'm wondering if we need to step back on this one. Should this be a feature request into core? As "Peter" the Plugin developer, I'd like a simple django model class to to inherit from,  that provides a consistent, tested and standardized way of enforcing a singleton pattern.  I am happy to help implement a standardized way into this project in a future PR as well.  Questions I have: The solution I provided, is it missing something?  If we recreate what John has shared, can someone explain what the pickled_data is doing? Sorry, I'm not sure I get that piece. Also, do we need to add a new dependency?  Not sure where ` DillPickledObjectField()` comes from.